addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobStore.java

```diff
@@ -128,7 +128,7 @@
     /** If true, include the object version in the key. */
     protected final boolean useVersion;
 
-    protected final boolean useAsyncDigest;
+    protected Boolean useAsyncDigest;
```

Comment:
The double-checked locking idiom requires the field to be `volatile` otherwise it's incorrect.